### PR TITLE
Wire up static-analysis so that it executes in the normal 'hack' flow

### PR DESCRIPTION
**Pull request by logicalhan**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

We have a stability framework to ensure API guarantees for metrics, but our static analysis runs in a blaze build/test which is unconventional. I've rewired it to expose a normal `hack/verify_*.sh`and `hack/update_*.sh` interface.

**Special notes for your reviewer**:

Ref: https://github.com/kubernetes/enhancements/blob/6c2e020d4475d4170ffdeafb266a84bb444ac9a6/keps/sig-instrumentation/1209-metrics-stability/20190404-kubernetes-control-plane-metrics-stability.md#api-review

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

Please use the following format for linking documentation:
- [KEP]: [metrics stability framework](https://github.com/kubernetes/enhancements/blob/6c2e020d4475d4170ffdeafb266a84bb444ac9a6/keps/sig-instrumentation/1209-metrics-stability/20190404-kubernetes-control-plane-metrics-stability.md#api-review)
- [Usage]: `./hack/verify-generated-stable-metrics.sh` & `./hack/update-generated-stable-metrics.sh`

/sig instrumentation

### CODE REVIEW

**Review by ehashman — APPROVED** [review]
Looks fine to me, deferring to SIG Testing

**Comment by ehashman** [comment]
@logicalhan pull-kubernetes-bazel-test is a test not found (legit) failure, can you PTAL?

File: test/instrumentation/stability-utils.sh [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+kube::validate::stablemetrics() {
+  stability_check_setup
+  static_checked_files=$(find_files_to_check | grep -E ".*.go" | grep -v ".*_test.go") || true
+  temp_file=$(mktemp)
+  gopathfiles=$(find_files_to_check | grep -E 'test/instrumentation/.*.go' | grep -v "test/instrumentation/main.*go" | cut -c 3-)
+  for i in "${static_checked_files[@]}"
+  do
+    # Deliberately allow word split here
+    # shellcheck disable=SC2086
+    stabilitycheck=$(go run "test/instrumentation/main.go" $gopathfiles -- $i 1>$temp_file)
```
Comment (liggitt) [inline test/instrumentation/stability-utils.sh:67]:
```
hack/verify-generated-stable-metrics.sh
go: gopkg.in/yaml.v2 upgrade => v2.4.0
hack/../test/instrumentation/stability-utils.sh: line 67: /Users/liggitt/.gvm/gos/go1.15/bin/go: Argument list too long
```

Comment (logicalhan) [reply test/instrumentation/stability-utils.sh]: Fine fine, I'll partition the invocations.

File: hack/update-generated-stable-metrics.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+# by default.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+source "${KUBE_ROOT}/test/instrumentation/stability-utils.sh"
+
+kube::update::stablemetrics
```
Comment (liggitt) [inline hack/update-generated-stable-metrics.sh:28]: I tried changing an existing metric from ALPHA to STABLE and running this script:

```diff
diff --git a/pkg/kubelet/certificate/kubelet.go b/pkg/kubelet/certificate/kubelet.go
index af2d21717dc..0c8ef588347 100644
--- a/pkg/kubelet/certificate/kubelet.go
+++ b/pkg/kubelet/certificate/kubelet.go
@@ -82,7 +82,7 @@ func NewKubeletServerCertificateManager(kubeClient clientset.Interface, kubeCfg
 				31104000,  // 1  year
 				124416000, // 4  years
 			},
-			StabilityLevel: compbasemetrics.ALPHA,
+			StabilityLevel: compbasemetrics.STABLE,
 		},
 	)
 	legacyregistry.MustRegister(certificateRotationAge)
```

```
$ hack/update-generated-stable-metrics.sh 
go: gopkg.in/yaml.v2 upgrade => v2.4.0
./pkg/kubelet/certificate/kubelet.go:70:4: Non string attribute it not supported
exit status 1
!!! [0208 16:47:02] Call tree:
!!! [0208 16:47:02]  1: hack/update-generated-stable-metrics.sh:28 kube::update::stablemetrics(...)

$ echo $?
1

$ git status
On branch so-stable
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   pkg/kubelet/certificate/kubelet.go

no changes added to commit (use "git add" and/or "git commit -a")
```

should that have updated `test/instrumentation/testdata/stable-metrics-list.yaml`?

Comment (logicalhan) [reply hack/update-generated-stable-metrics.sh:28]: Ah, yeah, Marek is tracking that one. There are some corner cases we don't support yet due (i.e. having a constant populating the subsystem field.

ref: https://github.com/kubernetes/kubernetes/issues/81863

**Comment by liggitt** [comment]
```
verify: generated-stable-metrics | 27s
```

/lgtm
/approve

**Pull request merged**